build-tools/packages/build-tools/src/repoPolicyCheck/handlers/npmPackages.ts

```diff
@@ -585,8 +586,102 @@ export const handlers: Handler[] = [
             return { resolved: resolved };
         },
     },
+    {
+        name: "npm-package-json-prettier",
+        match,
+        handler: (file) => {
+            let json;
+
+            try {
+                json = JSON.parse(readFile(file));
+            } catch (err) {
+                return "Error parsing JSON file: " + file;
+            }
+
+            const hasPrettierScript = Object.prototype.hasOwnProperty.call(
+                json.scripts,
+                "prettier",
+            );
+            const hasPrettierFixScript = Object.prototype.hasOwnProperty.call(
+                json.scripts,
+                "prettier:fix",
+            );
+            const hasFormatScript = Object.prototype.hasOwnProperty.call(json.scripts, "format");
+
+            if (!(hasPrettierScript && hasPrettierFixScript && hasFormatScript)) {
+                const missingScripts: string[] = [];
+
+                if (!hasPrettierScript) {
+                    missingScripts.push(`prettier`);
+                }
+
+                if (!hasPrettierFixScript) {
+                    missingScripts.push(`prettier:fix`);
+                }
+
+                if (!hasFormatScript) {
+                    missingScripts.push(`format`);
+                }
+
+                return `${file} is missing the following scripts: ${missingScripts.join("\n\t")}`;
+            }
+
+            return undefined;
+        },
+        resolver: (file) => {
+            let json;
+            try {
+                json = JSON.parse(readFile(file));
+            } catch (err) {
+                return { resolved: false, message: "Error parsing JSON file: " + file };
+            }
+
+            const resolved = true;
+
+            addPrettier(json);
+
+            writeFile(file, JSON.stringify(sortPackageJson(json), undefined, 2) + newline);
+
+            return { resolved: resolved };
+        },
+    },
 ];
 
+function addPrettier(json: Record<string, any>) {
+    const jsonCheckRoot = json.name;
+    let jsonDevDependencies = json.devDependencies;
+    let jsonScripts = json.scripts;
```

Comment:
Do we need the two local variables? Feels like the rest of the function should be able to check/update `json["property-name"]` in place.

Reply:
Both local variables removed.